sync/rpc/server.go

```diff
@@ -1214,3 +1214,63 @@ func (b *BackupList) BackupDelete(backupID string) error {
 	}
 	return fmt.Errorf("backup not found %v", backupID)
 }
+
+func (s *SyncAgentServer) FileSync(ctx context.Context, req *FileSyncRequest) (*empty.Empty, error) {
+	if err := s.syncFilesFromHealthyReplica(req.FromServer, req.Disks); err != nil {
+		return nil, err
+	}
+
+	return &empty.Empty{}, nil
+}
+
+func (s *SyncAgentServer) syncFilesFromHealthyReplica(fromServer string, disks []string) (err error) {
+	for _, disk := range disks {
+		if err := s.syncFile(fromServer, disk, ""); err != nil {
+			return err
+		}
+
+		if err := s.syncFile(fromServer, disk+".meta", ""); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (s *SyncAgentServer) syncFile(fromServer string, fromFileName, toFileName string) error {
+	if toFileName == "" {
+		toFileName = fromFileName
+	}
+
+	myHost := strings.Split(s.replicaAddress, ":")[0]
+	reply, err := s.ReceiverLaunch(nil, &ReceiverLaunchRequest{
+		ToFileName: toFileName,
+	})
+	if err != nil {
+		return err
+	}
+
+	myPort := reply.Port
+
+	conn, err := grpc.Dial(fromServer, grpc.WithInsecure())
+	if err != nil {
+		return fmt.Errorf("cannot connect to SyncAgentService %v: %v", fromServer, err)
+	}
+	defer conn.Close()
+	syncAgentFromClient := NewSyncAgentServiceClient(conn)
+
+	ctx, cancel := context.WithTimeout(context.Background(), GRPCServiceCommonTimeout)
```

Comment:
We don't know how long it will take to copy a file, it can be TB size. Timeout makes no sense.

Reply:
ok. Will use context.Cancel(context.Background()) instead.